app/src/pages/playground/schemas.ts

```diff
@@ -93,11 +95,80 @@ const chatMessageSchema = schemaForType<ChatMessage>()(
 export const chatMessagesSchema = z.array(chatMessageSchema);
 
 /**
- * The zod schema for llm model name
+ * The zod schema for llm model config
  * @see {@link https://github.com/Arize-ai/openinference/blob/main/spec/semantic_conventions.md|Semantic Conventions}
  */
-export const modelNameSchema = z.object({
+export const modelConfigSchema = z.object({
   [SemanticAttributePrefixes.llm]: z.object({
     [LLMAttributePostfixes.model_name]: z.string(),
+    [LLMAttributePostfixes.invocation_parameters]: z
+      .string()
+      .transform((s) =>
+        z
+          // TODO(apowell): specify schema for parameters that we care about
+          // convert them to camelCase after parsing so that we don't have to convert
+          // before hitting zustand
+          .object({})
+          .parse(JSON.parse(s))
+      )
+      .default("{}"),
   }),
 });
+
+/**
+ * Model invocation parameters schema in zod.
+ *
+ * Includes all keys besides toolChoice
+ */
+const invocationParameterSchema = schemaForType<
+  Mutable<InvocationParameters>
+>()(
+  z.object({
+    temperature: z.number().optional(),
+    topP: z.number().optional(),
+    maxTokens: z.number().optional(),
+    stop: z.array(z.string()).optional(),
+    seed: z.number().optional(),
+    maxCompletionTokens: z.number().optional(),
+  })
+);
+
+export type InvocationParametersSchema = z.infer<
+  typeof invocationParameterSchema
+>;
+
+/**
+ * Default set of invocation parameters for all providers and models.
+ */
+const baseInvocationParameterSchema = invocationParameterSchema.omit({
+  maxCompletionTokens: true,
+});
+
+/**
+ * Invocation parameters for O1 models.
+ */
+const o1BaseInvocationParameterSchema = invocationParameterSchema.pick({
+  maxCompletionTokens: true,
+});
+
+/**
+ * Provider schemas for all models and optionally for a specific model.
+ */
+export const providerSchemas = {
+  OPENAI: {
+    default: baseInvocationParameterSchema,
+    "o1-preview": o1BaseInvocationParameterSchema,
+    "o1-preview-2024-09-12": o1BaseInvocationParameterSchema,
+    "o1-mini": o1BaseInvocationParameterSchema,
+    "o1-mini-2024-09-12": o1BaseInvocationParameterSchema,
+  },
+  AZURE_OPENAI: {
+    default: baseInvocationParameterSchema,
+  },
+  ANTHROPIC: {
+    default: baseInvocationParameterSchema,
+  },
+} satisfies Record<
```

Comment:
nice, forced em to go back and read about satisfies, this is great

Reply:
It's so good once it clicks. You get inference similar to "as const" against some type without all the readonly garbage